policyengine_us/variables/gov/states/pa/tax/income/pa_nontaxable_pension_income.py

```diff
@@ -0,0 +1,16 @@
+from policyengine_us.model_api import *
+
+
+class pa_nontaxable_pension_income(Variable):
+    value_type = float
+    entity = Person
+    label = "Pension income taxable by US but not by PA"
+    unit = USD
+    documentation = "US taxable pension income excluded from PA AGI."
+    definition_period = YEAR
+    reference = "https://www.revenue.pa.gov/FormsandPublications/FormsforIndividuals/PIT/Documents/2021/2021_pa-40in.pdf#page=8"
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        us_taxable_pension = person("taxable_pension_income", period)
+        return where(age >= 65, us_taxable_pension, 0)
```

Comment:
Could you instead add an `is_retired` variable and call that here? It can live in `household/demographic/person/`.

That variable can then have a formula of `(age >= 65) & (earned_income == 0)`, with a comment that this is essentially an imputation.

Reply:
The `(earned_income == 0)` part will not work.  All pensions and even social security for 65+ allow positive earnings in retirement without any reduction in benefits.  I really don't see any advantage to your suggestion, especially because other states may have a different rule.   So, having this rule as part of the `pa_nontaxable_pension_income` variable formula is a big advantage.

Reply:
What about a new parameter named `pa_pension_exclusion_age` which would be set to 65, and then used instead of 65 in the existing formula for the `pa_nontaxable_pension_income` variable?

Reply:
OK we can take out the earnings requirement. But to confirm, PA law doesn't specify age 65 right? If that's the case, let's keep the PA formulas reflecting PA law, and move assumptions elsewhere; each state-specific policy parameters should reflect laws of that state. If another state defines retirement more precisely, we can use that more precise definition in its code.

Reply:
@MaxGhenis said:
> But to confirm, PA law doesn't specify age 65 right?

That's right, the instructions on the PA state income tax form do not define an age.  I have no idea what, if anything, the PA tax legislation says about this.

Given that information, do you want me to change the PR in any way?  
If so, how should I change it?

Reply:
Yes please 
1. Create a new is_retired variable in household/demographics/person
2. Set is_retired's formula to age>=65 and add a comment that this is a semi-arbitrary imputation 
3. Change this formula to call is_retired

Reply:
I've done as you asked.